From 8351d1f0ce814dfe4ec0919b5bba844ec068d0f3 Mon Sep 17 00:00:00 2001
From: Banka Sai Krishna <quic_banksaik@quicinc.com>
Date: Fri, 6 Dec 2024 16:14:37 +0530
Subject: [PATCH] QSDK:CP:wdev.uc: Add multi-radio monitor mode config

Add configuration for multi-radio monitor mode with center_frequency and bandwidth settings.

Change-Id: I8fa77d01ed26820b2a1b6f2dbddfd0aa49c4516c
Signed-off-by: Banka Sai Krishna <quic_banksaik@quicinc.com>
---
 .../wifi-scripts/files/usr/share/hostap/wdev.uc    | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/package/network/config/wifi-scripts/files/usr/share/hostap/wdev.uc b/package/network/config/wifi-scripts/files/usr/share/hostap/wdev.uc
index 89255dc9fda3..c313b93e2bd1 100644
--- a/package/network/config/wifi-scripts/files/usr/share/hostap/wdev.uc
+++ b/package/network/config/wifi-scripts/files/usr/share/hostap/wdev.uc
@@ -33,8 +33,18 @@ function iface_start(wdev)
 	phydev.wdev_add(ifname, wdev_config);
 	wdev_set_up(ifname, true);
 	let htmode = wdev.htmode || "NOHT";
-	if (wdev.freq)
-		system(`iw dev ${ifname} set freq ${wdev.freq} ${htmode}`);
+	if (wdev.is_multi_radio != null) {
+		if (wdev.mode == "monitor") {
+			if (wdev.center_freq) {
+				system(`iw dev ${ifname} add freq ${wdev.freq} ${wdev.bw} ${wdev.center_freq}`);
+			} else {
+				system(`iw dev ${ifname} add freq ${wdev.freq} ${wdev.bw}`);
+			}
+		}
+	} else {
+		if (wdev.freq)
+			system(`iw dev ${ifname} set freq ${wdev.freq} ${wdev.htmode}`);
+	}
 	if (wdev.mode == "adhoc") {
 		let cmd = ["iw", "dev", ifname, "ibss", "join", wdev.ssid, wdev.freq, htmode, "fixed-freq" ];
 		if (wdev.bssid)
-- 
2.34.1

